src/commands/api/findTreeItem.ts

```diff
@@ -83,34 +72,44 @@ export async function findTreeItem(query: TreeItemQuery): Promise<DatabaseAccoun
-            return undefined;
-        }
+    try {
+        for (const dbAccount of dbAccounts) {
+            if (Date.now() > maxTime) {
+                return undefined;
+            }
 
-        let actual: ParsedConnectionString;
-        if (dbAccount instanceof MongoAccountTreeItem) {
-            actual = await parseMongoConnectionString(dbAccount.connectionString);
-        } else if (dbAccount instanceof DocDBAccountTreeItemBase) {
-            actual = parseDocDBConnectionString(dbAccount.connectionString);
-        } else {
-            return undefined;
-        }
+            let actual: ParsedConnectionString;
+            if (dbAccount instanceof MongoAccountTreeItem) {
+                actual = await parseMongoConnectionString(dbAccount.connectionString);
+            } else if (dbAccount instanceof DocDBAccountTreeItemBase) {
+                actual = parseDocDBConnectionString(dbAccount.connectionString);
+            } else {
+                return undefined;
+            }
 
-        if (expected.accountId === actual.accountId) {
-            if (expected.databaseName) {
-                const dbs = await dbAccount.getCachedChildren(context);
-                for (const db of dbs) {
-                    if ((db instanceof MongoDatabaseTreeItem || db instanceof DocDBDatabaseTreeItemBase) && expected.databaseName === db.databaseName) {
-                        return new DatabaseTreeItemInternal(expected, dbAccount, db);
+            if (expected.accountId === actual.accountId) {
+                if (expected.databaseName) {
+                    const dbs = await dbAccount.getCachedChildren(context);
+                    for (const db of dbs) {
+                        if ((db instanceof MongoDatabaseTreeItem || db instanceof DocDBDatabaseTreeItemBase) && expected.databaseName === db.databaseName) {
+                            return new DatabaseTreeItemInternal(expected, dbAccount, db);
+                        }
                     }
+
+                    // We found the right account - just not the db. In this case we can still 'reveal' the account
+                    return new DatabaseTreeItemInternal(expected, dbAccount);
                 }
 
-                // We found the right account - just not the db. In this case we can still 'reveal' the account
-                return new DatabaseTreeItemInternal(expected, dbAccount);
+                return new DatabaseAccountTreeItemInternal(expected, dbAccount);
             }
-
-            return new DatabaseAccountTreeItemInternal(expected, dbAccount);
+        }
+    } catch (error) {
+        const parsedError: IParsedError = parseError(error);
+        if (parsedError.errorType === 'MongoNetworkError' && !parsedError.message.includes(expected.accountId)) {
+            // Ignore this error since it doesn't pertain to the account we're searching for
+            // https://github.com/microsoft/vscode-cosmosdb/issues/966
+        } else {
+            throw error;
```

Comment:
If you throw the error, we will have this same issue for when users have SQL Core or Postgres attached accounts that fail.  As of now, this only fixes an errors that Mongo throws.

I think you can just check if the expected accountId is included in the error and throw the error then.  If it's not, then you can return undefined in the catch block because we don't want unrelated errors to screw up the Connections node.

Reply:
![image](https://user-images.githubusercontent.com/5290572/77194498-52377f80-6a9d-11ea-92b5-53cd6d38b7c2.png)
 LOL uh-oh.  Got the anger sharks swimming in my head!!

Reply:
I feel like you guys might be nervous to swallow all errors here because I usually make comments like [this](https://github.com/microsoft/vscode-azurestorage/pull/618#discussion_r391340142) telling you to check `parsedError.errorType`, but sometimes there's exceptions to that rule. For example [here](https://github.com/microsoft/vscode-azurefunctions/blob/master/src/templates/script/parseScriptTemplates.ts#L242). I think it's fine to swallow all errors in this case since you don't want _any_ error to block you from searching other dbAccounts